src/test/java/teammates/test/cases/datatransfer/FeedbackMcqQuestionDetailsTest.java

```diff
@@ -0,0 +1,288 @@
+        assertEquals(0.0, mcqDetails.getMcqOtherWeight());
+        requestParams.clear();
+
+        ______TS("Failure: weights enabled, two choices, null weights, other disabled");
+        mcqDetails = new FeedbackMcqQuestionDetails();
+
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_TYPE, new String[] { "MCQ" });
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_TEXT, new String[] { "mcq question text" });
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_MCQ_GENERATED_OPTIONS, new String[] { "NONE" });
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_NUMBEROFCHOICECREATED, new String[] { "2" });
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_MCQCHOICE + "-0", new String[] { "Choice 1" });
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_MCQCHOICE + "-1", new String[] { "Choice 2" });
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_MCQ_WEIGHTS_ASSIGNED, new String[] { "on" });
+
+        assertTrue(mcqDetails.extractQuestionDetails(requestParams, FeedbackQuestionType.MCQ));
+        assertTrue(mcqDetails.hasAssignedWeights());
+        assertFalse(mcqDetails.getMcqChoices().isEmpty());
+        assertEquals(2, mcqDetails.getMcqChoices().size());
+        weights = mcqDetails.getMcqWeights();
+        assertTrue(weights.isEmpty());
+        assertEquals(0, weights.size());
+        requestParams.clear();
+
+        ______TS("Success: Weights enalbed, two choices, custom weights, other disabled");
+        mcqDetails = new FeedbackMcqQuestionDetails();
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_TYPE, new String[] { "MCQ" });
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_TEXT, new String[] { "mcq question text" });
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_MCQ_GENERATED_OPTIONS, new String[] { "NONE" });
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_NUMBEROFCHOICECREATED, new String[] { "2" });
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_MCQCHOICE + "-0", new String[] { "Choice 1" });
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_MCQCHOICE + "-1", new String[] { "Choice 2" });
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_MCQ_WEIGHTS_ASSIGNED, new String[] { "on" });
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_MCQ_WEIGHT + "-0", new String[] { "1.50" });
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_MCQ_WEIGHT + "-1", new String[] { "2.50" });
+
+        assertTrue(mcqDetails.extractQuestionDetails(requestParams, FeedbackQuestionType.MCQ));
+        assertTrue(mcqDetails.hasAssignedWeights());
+        assertFalse(mcqDetails.getMcqChoices().isEmpty());
+        assertEquals(2, mcqDetails.getMcqChoices().size());
+        weights = mcqDetails.getMcqWeights();
+        assertFalse(weights.isEmpty());
+        assertEquals(2, weights.size());
```

Comment:
I think asserting `isEmpty` is redundant?

Reply:
I guess you are right. I was trying to be extra cautious I guess :laughing: I guess only checking the size is okay then.